### DISCUSSION TITLE
HOW TO MAKE A REACH CONTRACT RECOGNIZE AND USE A NON-NETWORK TOKEN LAUNCHED OUTSIDE THE CONTRACT

### DISCUSSION THREAD

**Discussion by kcelestinomaria** [reach-sh/reach-lang#1060] [Q&A] [answered]
Hello Reach Fam,

There is something I am trying to figure out. I have not been able to see any Reach examples where the Reach program lets the participants interact with a non-network token that was launched outside the Reach Contract, and is also not a test token. 

I am still looking for an example or solution. I want the program(from its frontend - .mjs file and RPC) to read a real token's(ASA) info from the network, without a participant being required to provide that token or any token info. This then allows a second participant to provide some amount of this ASA to the contract, and then the contract sends an amount of this ASA to the first participant once this first participant provides the amount he wants.

Is this possible? Could sm help here? Thanks

**Comment by jeapostrophe** [comment, marked as answer]
"I want the program to read a real token's info from the network, without a participant being required to provide that token or any token info" 

This is not possible. Programs cannot learn anything that doesn't come from participants. You need to have a participant publish a `Token` value. That value doesn't have to be a "test" token. You could give it `31566704` if you wanted to use USDC --- https://algoexplorer.io/asset/31566704

**Comment by kcelestinomaria** [comment]
What about on the frontend? How would I declare the USDC Token, for example, or some new token I created using the Reach standard library? @jeapostrophe

**Reply by jeapostrophe** [reply]
If you want to create a token fresh, then you can use `stdlib.launchToken` --- https://docs.reach.sh/frontend/#launching-tokens

If you want to create a value that represents a `Token`, then it depends on the connector --- https://docs.reach.sh/frontend/#p_6 --- in the case of ALGO, it is just a number, so you'd use `31566704` for USDC.

**Reply by kcelestinomaria** [reply]
Is the ASA ID recognize as an integer/number or as a string?

**Reply by kcelestinomaria** [reply]
So basically ```const usdctokenid = 31566704; 
await accAlice.tokenAccept(usdctokenid)``` would work, right?

**Reply by jeapostrophe** [reply]
Yup

**Comment by kcelestinomaria** [comment]
Is there like a standard schema e.g ```const newToken``` or ```const usdcToken(if I am the creator) = stdlib.getTokenSchemaFromNetwork({ name: usdcToken, url: 'usdc.org', id: '31566704' })``` @jeapostrophe

**Reply by jeapostrophe** [reply]
I don't understand... you just write `const usdc = 31566704;`... If you want to download information about it... that's between you and USDC. There's no standard way to talk about assets on Algorand other than by these numbers

**Reply by kcelestinomaria** [reply]
I understand now, I thought there was a standard way after seeing the ```.stdlib.launchToken()``` examples. Thanks! @jeapostrophe
